.github/workflows/iroha2-dev.yml

```diff
@@ -23,71 +27,48 @@ jobs:
         id: buildx
         with:
           install: true
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build and push docker image
-        uses: docker/build-push-action@v2
+          driver: docker
+
+      - name: Build iroha2:env container
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:env
           context: .
-          push: true
-          tags: hyperledger/iroha2:dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-      - name: Build and push docker image (iroha_client_cli)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.ci        
+
+      - name: Build iroha2:build docker image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:build
           context: .
-          push: true
-          tags: hyperledger/iroha2:client-cli-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=iroha_client_cli
-      - name: Build and push docker image (kagami)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.build
+
+      - name: build iroha2:dev docker image
+        run: |
+          docker build -f Dockerfile -t hyperledger/iroha2:dev .
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
         with:
-          context: .
-          push: true
-          tags: hyperledger/iroha2:kagami-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=kagami
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: push iroha2:dev docker image
+        run: |
+          docker push hyperledger/iroha2:dev
```

Comment:
There's also non-negligible overhead. The actions assume that docker is not installed and not configured. This flies in the face of including the entire infrastructure at the level of the `iroha2:env` image.

Reply:
I suggest to move this issue to another new PR. So, to create a new ticket for DevOps team and a corresponding issue for this task. It will be combined with a new workflow file for `iroha2:env (base)` image building.